packages/astro/test/astro-pagination.test.js

```diff
@@ -41,12 +41,28 @@ describe('Pagination', () => {
 			{ color: 'blue', p: '2' },
 		];
 		await Promise.all(
-			params.map(async ({ color, p }) => {
+			params.map(async ({ color, p }, idx) => {
 				const html = await fixture.readFile(`/posts/${color}/${p}/index.html`);
 				const $ = cheerio.load(html);
-				expect($('#page-a').text()).to.equal(p);
-				expect($('#page-b').text()).to.equal(p);
+				expect($('#page-param').text()).to.equal(p);
+				expect($('#currentPage').text()).to.equal(p);
 				expect($('#filter').text()).to.equal(color);
+
+				const prevHref = $('#prev').attr('href');
+				const nextHref = $('#next').attr('href');
+
+				if (color === 'red') {
+					expect(prevHref).to.be.undefined;
+					expect(nextHref).to.be.undefined;
+				}
+				if (color === 'blue' && p === 1) {
+					expect(prevHref).to.be.undefined;
+					expect(nextHref).to.equal('/posts/blue/2');
+				}
+				if (color === 'blue' && p === 2) {
+					expect(prevHref).to.equal('/posts/blue/1');
+					expect(nextHref).to.be.undefined;
```

Comment:
Do these tests every run? Say when `p = '1'`  ===> `p  ===  1` is false which will cause the test to not run

Reply:
Sorry, this is a super imperative way to check if the `params` entry matches one of the items in the list ([see the lines just above](https://github.com/withastro/astro/pull/6676/files#diff-febabdfee1785d50a08874c42cbcb3d8eb6e6f420fb37843dcdce0511cb6f72fR39-R41)). Brain-dead solution I know, but oddly the simplest I could find 😅

Reply:
I meant that `p` is always a string so comparing it the numbers will return `false`
'1' !=== 1

I'm I overlooking something?

Reply:
Ohhhh right you are! I'm a fool. Will fix